### PR TITLE
added data_manager.py

### PR DESCRIPTION
**Pull request by arturiermolenko**
added first version of data_manager with basic authentication

### CODE REVIEW

File: src/data_manager.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        Load the data from a pickle file or to create a new AddressBook object.
+        """
+        try:
+            with open(self.filename, "rb") as f:
+                return pickle.load(f)
+        except FileNotFoundError:
+            return AddressBook()
+
+    @staticmethod
+    def generate_filename(email: str, password: str) -> str:
```
Comment (payemo) [inline src/data_manager.py]: Rename method to `generate_hash`

**Review by any-nick — APPROVED** [review]
Reviewed

File: src/data_manager.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    def login(self):
+        """
+        Input of email and password. If such pair already exists - create and return filename.
+        Through to main menu if it does not exist.
+        :return: hashed email + password
+        """
+        email = input("Enter email: ")
+        pwd = input("Enter password: ")
+        file_name = self.generate_filename(email, pwd)
```
Comment (payemo) [inline src/data_manager.py]: Rename the variable to `secret_hash`

File: src/data_manager.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
+import hashlib
+import os
+import pickle
+import sys
+from collections import UserDict
+from models import AddressBook
+
+
+class DataManager:
```
Comment (payemo) [inline src/data_manager.py:9]: - Rename to `DataSerializer`
- Use `DataSerializer` class specifically for serializing data: leave `load_data` and `save_data` methods
- Move `generate_hash/login/signup/validate_secret` to another class - `AuthManager`
- Don't use ``

File: src/data_manager.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        """
+        Generate a filename based on the email and password.
+        :param email: user`s email from input
+        :param password: user`s password from input
+        :return:
+        """
+        hash_obj = hashlib.sha256(email.encode() + password.encode())
+        return hash_obj.hexdigest()
+
+    def input_correct_passwords(self) -> str:
```
Comment (payemo) [inline src/data_manager.py]: Rename to `validate_secret`

File: src/data_manager.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        Taking 2 passwords from input and check if they are same.
+        :return: Password(str)
+        """
+        pwd = input("Enter password: ")
+        conf_pwd = input("Confirm password: ")
+        if conf_pwd == pwd:
+            return pwd
+        else:
+            print("Passwords are not identical! Try one more time please.\n")
+            self.input_correct_passwords()
```
Comment (payemo) [inline src/data_manager.py]: Don't use recursion in such a way: make a `for in range(3)` in `signup` method

File: src/data_manager.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        file_name = self.generate_filename(email, pwd)
+
+        if not os.path.exists(file_name):
+            print("User with such email or password does not exist!")
+            file_name = self.get_filename()
+        else:
+            print("Logged in Successfully!")
+        return file_name
+
+    def get_filename(self) -> bool:
```
Comment (payemo) [inline src/data_manager.py:84]: It's too much actions for method called `get_file_name`, rename - `auth`

**Review by payemo — APPROVED** [review]

**Pull request closed without merging**